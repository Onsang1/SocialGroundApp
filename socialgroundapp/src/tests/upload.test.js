/**
* @jest-environment jsdom
*/
import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
// import testing library functions
// import userEvent from '@testing-library/user-event';
import { expect, test } from '@jest/globals';
import renderer from 'react-test-renderer';
// import { within } from '@testing-library/dom';
import '@testing-library/jest-dom/extend-expect';
import Upload from '../components/upload';

test('renders Upload link', () => {
  render(
    <Router>
      <Upload />
    </Router>,
  );
  const linkElement = screen.getByText(/Share/);
  expect(linkElement).toBeInTheDocument();
});

test('renders Upload link', () => {
  render(
    <Router>
      <Upload />
    </Router>,
  );
  const linkElement = screen.getByText(/Cancel/);
  expect(linkElement).toBeInTheDocument();
});

// snapshot testing

test('SocialGround upload page matches snapshot', () => {
  const component = renderer.create(
    <Router>
      <Upload />
    </Router>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// const setup = () => {
//   const utils = render(
//     <Router>
//       <Upload />
//     </Router>,
//   );
//   const input = utils.getByTestId('fileInput');
//   return {
//     input,
//     ...utils,
//   };
// };

// test('click', () => {
//   const testFile = new File(['test'], 'test.png', { type: 'image/png' });
//   const { input } = setup();
//   fireEvent.change(input, { target: { file: testFile } });
//   expect(input.file).toBe(testFile);
// });
