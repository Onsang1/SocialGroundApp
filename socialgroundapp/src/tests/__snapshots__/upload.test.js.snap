// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialGround upload page matches snapshot 1`] = `
<div
  id="react-root"
>
  <div
    aria-disabled="false"
    role="button"
  >
    <a
      href="/Feed"
      onClick={[Function]}
    >
      <img
        alt="sg-logo-small"
        className="sg-logo-small"
        src={Object {}}
      />
    </a>
  </div>
  <div
    className="main-body"
  >
    <div
      className="select-box"
    />
    <div
      className="select-text"
    >
      <form
        action="http://localhost:3000.com/"
        method="POST"
      >
        <img
          alt="Select file from device"
          id="imageid"
        />
        <input
          onChange={[Function]}
          type="file"
        />
      </form>
    </div>
    <span
      className="new-post"
    >
      New Post
    </span>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <span
        className="new-post"
      >
        New Post
      </span>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <button
        className="share"
        type="button"
      >
        Share
      </button>
    </a>
    <a
      href="/profile"
      onClick={[Function]}
    >
      <span
        className="cancel"
      >
        Cancel
      </span>
    </a>
    <blockquote
      className="caption-box"
      contentEditable="true"
    >
      <h1
        style={
          Object {
            "color": "gray",
          }
        }
      >
        Enter Caption...
      </h1>
    </blockquote>
    <div
      className="footnotes"
    >
      <span>
        @ Tag Users
      </span>
      <br />
      <span>
        Privacy Settings
      </span>
    </div>
  </div>
</div>
`;
